File: app/src/main/java/com/example/mlallemant/mentalbattle/UI/Game/Fragment/PlayerFindFragment.java
package com.example.mlallemant.mentalbattle.UI.Game.Fragment;

import android.app.Activity;
import android.app.Fragment;
import android.content.Context;
import android.os.Build;
import android.os.Bundle;
import android.os.CountDownTimer;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.mlallemant.mentalbattle.R;
import com.example.mlallemant.mentalbattle.Utils.Utils;

/**
 * Created by m.lallemant on 18/10/2017.
 */

public class PlayerFindFragment extends Fragment {

    OnCountdownFinish mCallBack;

    public interface OnCountdownFinish {
        void launchGame();
    }

    @Override
    public View onCreateView(final LayoutInflater inflater, final ViewGroup container, final Bundle savedInstanceState) {
        final View v = inflater.inflate(R.layout.player_find_fragment, container, false);

        final Bundle args = getArguments();
        final String currentPlayerName = args.getString("currentPlayer");
        final String otherPlayerName = args.getString("otherPlayer");

        final TextView tvPlayerOne = (TextView) v.findViewById(R.id.tv_player1);
        final TextView tvPlayerTwo = (TextView) v.findViewById(R.id.tv_player2);

        tvPlayerOne.setText(currentPlayerName.split(" ")[0]);
        tvPlayerTwo.setText(otherPlayerName.split(" ")[0]);

        launchCountDown(v);

        return v;
    }


    @Override
    public void onAttach(final Activity activity) {
        super.onAttach(activity);
        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.LOLLIPOP_MR1) return;
        if (activity instanceof OnCountdownFinish) {
            mCallBack = (OnCountdownFinish) activity;
        } else {
            throw new RuntimeException(activity.toString()
                    + " must implement OnListFragmentInteractionListener");
        }
    }

    @Override
    public void onAttach(final Context context) {
        super.onAttach(context);
        if (context instanceof OnCountdownFinish) {
            mCallBack = (OnCountdownFinish) context;
        } else {
            throw new RuntimeException(context.toString()
                    + " must implement OnListFragmentInteractionListener");
        }
    }

    private void launchCountDown(final View v) {

        final TextView tv_countdown = (TextView) v.findViewById(R.id.tv_countdown);
        new CountDownTimer(Utils.COUNTDOWN_LOBBY, 1000) {
            public void onTick(final long millisUntilFinished) {

                final String remainingTime = "" + millisUntilFinished / 1000;
                tv_countdown.setText(remainingTime);
            }

            public void onFinish() {
                mCallBack.launchGame();
            }

        }.start();
    }
}

File: app/src/main/java/com/example/mlallemant/mentalbattle/ui/game/fragment/PlayerFindFragment.kt
package com.example.mlallemant.mentalbattle.ui.game.fragment

import android.content.Context
import android.os.Bundle
import android.os.CountDownTimer
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.example.mlallemant.mentalbattle.databinding.PlayerFindFragmentBinding
import com.example.mlallemant.mentalbattle.ui.game.GameActivity.Companion.BUNDLE_EXTRA_CURRENT_PLAYER
import com.example.mlallemant.mentalbattle.ui.game.GameActivity.Companion.BUNDLE_EXTRA_OTHER_PLAYER
import com.example.mlallemant.mentalbattle.utils.Utils

/**
 * Created by m.lallemant on 18/10/2017.
 */
class PlayerFindFragment : Fragment() {

    private var _binding: PlayerFindFragmentBinding? = null
    private val binding get() = _binding!!

    var mCallBack: OnCountdownFinish? = null

    interface OnCountdownFinish {
        fun launchGame()
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
       _binding = PlayerFindFragmentBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        val currentPlayerName = arguments?.getString(BUNDLE_EXTRA_CURRENT_PLAYER)
        val otherPlayerName = arguments?.getString(BUNDLE_EXTRA_OTHER_PLAYER)
        binding.tvPlayer1.text = currentPlayerName!!.split(" ".toRegex()).toTypedArray()[0]
        binding.tvPlayer2.text = otherPlayerName!!.split(" ".toRegex()).toTypedArray()[0]
        launchCountDown()
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        mCallBack = if (context is OnCountdownFinish) {
            context
        } else {
            throw RuntimeException("$context must implement OnListFragmentInteractionListener")
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        _binding = null
    }

    private fun launchCountDown() {
        object : CountDownTimer(Utils.COUNTDOWN_LOBBY.toLong(), 1000) {

            override fun onTick(millisUntilFinished: Long) {
                val remainingTime = "" + millisUntilFinished / 1000
                binding.tvCountdown.text = remainingTime
            }

            override fun onFinish() {
                mCallBack?.launchGame()
            }
        }.start()
    }
}